notebooks/python/neo4j/README.md

```diff
@@ -0,0 +1,253 @@
+
+## Comparing APIs
+The examples exercise several pythonic APIs.  Which one should you use?  As shown in the truth table below, the GDS client is the obvious choice for GDS projects that don’t use Spark.  
+
+![API selector truth table](images/neo4j_dataproc_api_selector.png)
+
+The GDS client API allows executing GDS function calls natively as well as any Cypher statement using ``run_cypher``.  Calling GDS natively function uses slightly different object syntax than writing out Cypher strings using the gds client.
+
+When working distributed data on Spark cluster, the Neo4j Spark Connector provides read and write support like other Spark Connectors.  Although the Spark Connector supports GDS calls using the Cypher ``CALL`` syntax, the best practice is to instantiate the GDS or python client as well to handle script calls more elegantly.
+
+## Using the GDS client
+The GDS api is Neo4j's newest, most elegant and compact python API.  Usage requires importing a single library and instantiating an object on one line.
+
+    from graphdatascience import GraphDataScience
+
+    neo4j_url = "bolt://<server>:7687"
+    neo4j_user = "<user>"
+    neo4j_password = "<password>"
+
+    gds = GraphDataScience(neo4j_url, auth=(neo4j_user, neo4j_password))
+
+Thereafter, executing Cypher statements of any kind simply requires calling:
+
+    gds.run_cypher('<Cypher statement>')
+
+For GDS functions, invocation is fluent such as:
+
+    g_msa, result = gds.graph.project(
+        'msa-graph',
+        'MSA',
+        '*',
+        nodeProperties = [
+            "logPopulation",
+            "logMedianHouseholdIncome",
+            "logMedianHomePrice",
+            "logPercentOver25WithBachelors"])
+
+In this case, `g_msa` is a handle to the in-memory projection.
+
+Writing data back from the in-memory graph is simple:
+
+    gds.graph.writeRelationship(
+        g_msa,
+        "IS_SIMILAR",
+        relationship_property="similarity"
+    )
+
+Return data structures from the GDS client are well documented.
+
+## Spark Connector client
+On DataProc in the current release, global variables applied in `SparkSession.builder` such as `neo4j.url` do not stick between method invocations.
+
+To read records (in parallel) from Neo4j into a Spark DataFrame, execute a command like this:
+
+    (neo4jDf=spark.read.format("org.neo4j.spark.DataSource")
+        .option("labels", "MSA")
+        .option("url", neo4j_url)  
+        .option("authentication.type", "basic")
+        .option("authentication.basic.username", neo4j_user)
+        .option("authentication.basic.password", neo4j_password)
+        .load())
+
+Writing records (single threaded) to Neo4j from a Spark DataFrame can be executed like this:
+
+    (castBigqueryDf.write
```

Comment:
Please surround with code block, trailing back-slashes are needed for multi-line statement.

Also - why is the statement surrounded with parenthesis?